### PR TITLE
#117 - extend TempFile and TempDirectory class

### LINKED ISSUE #117: Extend Temp File and Directory classes
Extend TempFile and TempDirectory classes to get stable outputs when using on different Operation Systems.
On Windows additional actions are needed to reach code behavior similar to unix.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- added `toString` and `toURI` methods

Closes #117

### CODE REVIEW

File: src/test/scala/za/co/absa/commons/io/TempDirectorySpec.scala [hunk 1]
```diff
@@ -45,6 +45,21 @@ class TempDirectorySpec extends AnyFlatSpec with Matchers {
     name3 should (startWith("foo") and endWith("bar"))
   }
 
+  it should "return tempDirectory as string" in {
```
Comment (wajda) [inline src/test/scala/za/co/absa/commons/io/TempDirectorySpec.scala:48]: In this context `it` is a `apply` method, while the scenario is about the `toString` method.
Also the scenario doesn't actually verify the given statement, but other unrelated statements.

File: src/test/scala/za/co/absa/commons/io/TempFileSpec.scala [hunk 2]
```diff
@@ -40,4 +41,20 @@ class TempFileSpec extends AnyFlatSpec with Matchers {
     name2 should endWith("bar")
     name3 should (startWith("foo") and endWith("bar"))
   }
+
+  it should "return tempFile as string" in {
```
Comment (wajda) [inline src/test/scala/za/co/absa/commons/io/TempFileSpec.scala:45]: ditto

File: src/test/scala/za/co/absa/commons/io/TempDirectorySpec.scala (hunk 1 above)
Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/io/TempDirectorySpec.scala]: Fixed in c6675808ba03a8efcc7111030996fdde162b626a.

File: src/test/scala/za/co/absa/commons/io/TempFileSpec.scala (hunk 2 above)
Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/io/TempFileSpec.scala]: Fixed in c6675808ba03a8efcc7111030996fdde162b626a.

File: src/test/scala/za/co/absa/commons/io/TempFileSpec.scala [hunk 3]
```diff
@@ -42,13 +42,20 @@ class TempFileSpec extends AnyFlatSpec with Matchers {
...
-  it should "return tempFile as string" in {
-    val tempFilePath: String = TempFile().deleteOnExit().toString
+  behavior of "`toString`"
 
+  it should "return valid string path" in {
+    val tempFile = TempFile().deleteOnExit()
+    val tempFilePath: String = tempFile.toString
+    val expectedPath = tempFile.path.toAbsolutePath.toString.replace("\\", "/")
+
+    expectedPath should equal(tempFilePath)
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/io/TempFileSpec.scala]: Sure. My bad. Fixed in a4bf388b5e970fcdcfe4ba0f9f185ff81e25c62c.

File: src/test/scala/za/co/absa/commons/io/TempFileSpec.scala [hunk 4]
```diff
@@ -42,13 +42,20 @@ class TempFileSpec extends AnyFlatSpec with Matchers {
...
     new File(tempFilePath).exists() should be(true)
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/io/TempFileSpec.scala]: True. Removed additional checks. a4bf388b5e970fcdcfe4ba0f9f185ff81e25c62c

File: src/test/scala/za/co/absa/commons/io/TempFileSpec.scala [hunk 5]
```diff
@@ -42,13 +42,20 @@ class TempFileSpec extends AnyFlatSpec with Matchers {
...
     tempFilePath.contains("\\") should be(false)
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply src/test/scala/za/co/absa/commons/io/TempFileSpec.scala]: True. Removed redundant checks. a4bf388b5e970fcdcfe4ba0f9f185ff81e25c62c

**Review by wajda — APPROVED** [review]
LGTM

**Review by dk1844 — COMMENTED** [review]
Otherwise LGTM.

**Pull request merged**